### ISSUE TITLE
Overflow when adding durations during pop

### ISSUE THREAD

**Issue by saulshanabrook** [egraphs-good/egglog#240]
Opening a bug raised by @RobinKa in https://github.com/metadsl/egglog-python/issues/53 that sometimes when calling `pop` an overflow is raised by adding the durations. Relevent part of the traceback copied:

```
thread '<unnamed>' panicked at 'overflow when adding durations', library/core/src/time.rs:914:31

[...]

  21:     0x7f22ff5419e0 - <core::time::Duration as core::ops::arith::AddAssign>::add_assign::hf0b4e48d4fab5e6b
                               at /rustc/eb26296b556cef10fb713a38f3d16b9886080f26/library/core/src/time.rs:921:17
  22:     0x7f22ff399742 - egglog::RunReport::union_times::h38ef9263e9e7347a
  23:     0x7f22ff399812 - egglog::RunReport::union::h651a29516887af3d
  24:     0x7f22ff39ab81 - egglog::EGraph::pop::hec17bdef39ad741e
  25:     0x7f22ff3a372a - egglog::EGraph::run_command::h1a909c7d32e16c7c
  26:     0x7f22ff3a70b6 - egglog::EGraph::run_program::h3cdc3518c784f89b
```

This didn't use to happen, but may have been introduced by https://github.com/egraphs-good/egglog/pull/225 ?

**Comment by oflatt** [comment]
Good catch! Can you confirm this is fixed by #241?

**Comment by RobinKa** [comment]
Yep can confirm all my tests pass now, thanks for the quick fix!
